docs/sdk/src/reference_docs/development_environment_advice.rs

```diff
@@ -0,0 +1,95 @@
+//! # Development Environment Advice
+//!
+//! Large Rust projects are known for sometimes long compile times and sluggish dev tooling, and
+//! polkadot-sdk is no exception.
+//!
+//! This page contains some advice to improve your workflow when using common tooling.
+//!
+//! ## Rust Analyzer Configuration
+//!
+//! [Rust Analyzer](https://rust-analyzer.github.io/) is the defacto [LSP](https://langserver.org/) for Rust. Its default
+//! settings are fine for smaller projects, but not well optimised for polkadot-sdk.
+//!
+//! Below is a suggested configuration in Lua (can use with `neovim/nvim-lspconfig`)
+//! with comments describing the rationale for each setting:
+//!
+//! ```lua
+//! ["rust-analyzer"] = {
+//!   cargo = {
+//!     # Check feature-gated code
+//!     features = "all",
+//!     extraEnv = {
+//!       # Use a seperate target dir for Rust Analyzer. Helpful if you want to use Rust
+//!       # Analyzer and cargo on the command line at the same time.
+//!       ["CARGO_TARGET_DIR"] = "target/nvim-rust-analyzer",
+//!       # Skip building WASM, there is never need for it here
+//!       ["SKIP_WASM_BUILD"] = "1",
+//!       # Improve stability
+//!       ["CHALK_OVERFLOW_DEPTH"] = "100000000",
+//!     },
+//!   },
+//!   procMacro = {
+//!     # Don't expand some problematic proc_macros
+//!     ignored = {
+//!       ["async-trait"] = { "async_trait" },
+//!       ["napi-derive"] = { "napi" },
+//!       ["async-recursion"] = { "async_recursion" },
+//!       ["async-std"] = { "async_std" },
+//!     },
+//!   },
+//!   rustfmt = {
+//!     # Use nightly formatting.
+//!     # See the polkadot-sdk CI job that checks formatting for the current version used in
+//!     # polkadot-sdk.
+//!     extraArgs = { "+nightly-2023-11-01" },
+//!   },
+//! },
+//! ```
+//!
+//! and the same configuration in JSON for VSCode's `settings.json`:
+//!
+//! ```json
+//! {
+//!   "rust-analyzer.cargo.features": "all",
+//!   "rust-analyzer.cargo.extraEnv": {
+//!     "CARGO_TARGET_DIR": "target/vscode-rust-analyzer",
+//!     "SKIP_WASM_BUILD": "1",
+//!     "CHALK_OVERFLOW_DEPTH": "100000000"
+//!   },
+//!   "rust-analyzer.procMacro.ignored": {
```

Comment:
I have never used these, are they basically achieving the same thing (but in a more targeted way) as the second part of the one I use?
```
"rust-analyzer.diagnostics.disabled": [
  "unresolved-macro-call",
  "unresolved-proc-macro"
],
```

Reply:
I think these are different. I've never seen issues that required I disable those diagnostics :/

Reply:
Yea I've subbed in what you've written here in my setup. Hard to ensure that it catches all as I found they would sometimes just pop up in my editor when I opened a file but not in the output usually. Maybe I need to set '--all-targets --workspace' actually.

I'd prefer the way you've got here as it's less of a sweeping rule, and we can add any others we find to this array if they pop up.